lib/ddtrace/contrib/grape/endpoint.rb

```diff
@@ -187,6 +186,24 @@ def endpoint_run_filters(name, start, finish, id, payload)
 
           private
 
+          def api_view(api)
+            # If the API inherits from Grape::API in version >= 1.2.0
+            # then the API will be an instance and the name must be derived from the base.
+            # See https://github.com/ruby-grape/grape/issues/1825
+            if defined?(::Grape::API::Instance) && api <= ::Grape::API::Instance
+              api.base.to_s
+            else
+              api.to_s
+            end
+          end
+
+          def endpoint_expand_path(endpoint)
+            route_path = endpoint.options[:path]
+
+            parts = (endpoint.routes.first.namespace.split('/') + route_path).reject { |p| p.blank? || p.eql?('/') }
+            parts.join('/').prepend('/')
```

Comment:
With quite a few string operations here, I think it would be safer to `rescue` any errors that happen in this method and return a "safe" fallback value.

The value we had before was `payload[:endpoint].options[:path].join('/')`, which might be a safe start for a fallback, but I leave it to you, as your feedback as a production user of Grape is more valuable in this decision.

Also feel free to challenge my initial assumption that "this method might not be safe, thus we need a fallback".

Reply:
@marcotc I think I can mitigate these concerns by ensuring that a string is present for `endpoint.routes.first.namespace`. Thoughts?

Reply:
> @marcotc I think I can mitigate these concerns by ensuring that a string is present for endpoint.routes.first.namespace. Thoughts?

I think this makes sense 👍